lib/path/default.nix

```diff
@@ -336,6 +336,34 @@ in /* No rec! Add dependencies on this file at the top. */ {
               ${subpathInvalidReason path}''
       ) 0 subpaths;
 
+  /* Split a subpath into its path component strings.
+  Throw an error if the subpath isn't valid, see `lib.path.subpath.isValid`.
+  Note that the returned path component strings are not valid subpaths themselves.
```

Comment:
This is a bit unfortunate, because we decided to always prefix subpaths with `./`: https://github.com/NixOS/nixpkgs/tree/master/lib/path#leading-dots-for-relative-paths

This is the only reason why subpaths are not valid as components.

Anyways, it's already too late now, and it's not like a big problem, just a bit unfortunate :upside_down_face:

Reply:
I don't even think it's a minor problem. Those are clearly components, not subpaths. No ambiguity there.

Reply:
Oh this has a mistake in it. The resulting strings _are_ valid subpaths, they're just not normalised